src/uthash.h

```diff
@@ -105,6 +102,30 @@ typedef unsigned char uint8_t;
 #define uthash_expand_fyi(tbl)            /* can be defined to log expands   */
 #endif
 
+#ifndef HASH_NOMEM_OK
+#define HASH_NOMEM_OK 0
+#endif
+
+#if HASH_NOMEM_OK
+/* malloc failures can be recovered from */
+
+#ifndef uthash_fatal
+#define uthash_fatal(obj) do{}while(0)
```

Comment:
> The `uthash_oom` [you mean the thing called `uthash_fatal` in the current patch, right?] is to be called for every OOM (I'm talking for HASH_NOMEM_OK=1 case only here).

Can you walk me through how it would get called, in the `HASH_NOMEM_OK=1` case, if the user called `HASH_ADD_KEYPTR_BYHASHVALUE_INORDER` and the failure happened, let's say, while allocating the initial bloom filter in `HASH_BLOOM_MAKE`?  I see `HASH_BLOOM_MAKE` calls `HASH_MEM_FAILED` to set `mem_ok=0`, but I don't see anyone calling `uthash_fatal` based on that.